compiler/native/compile_test.go

```diff
@@ -1453,13 +1469,21 @@ func Test_client_modifyConfig(t *testing.T) {
 
 	engine.POST("/config/unmodified", func(c *gin.Context) {
 		c.Header("Content-Type", "application/json")
-		c.JSON(http.StatusOK, want)
+		response, err := convertResponse(want)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		}
+		c.JSON(http.StatusOK, response)
 	})
 
 	engine.POST("/config/timeout", func(c *gin.Context) {
 		time.Sleep(3 * time.Second)
 		c.Header("Content-Type", "application/json")
-		c.JSON(http.StatusOK, want)
+		response, err := convertResponse(want)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		}
+		c.JSON(http.StatusOK, response)
```

Comment:
if we dont return after c.JSON() will a double c.JSON() cause a double-write error?
```suggestion
			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
			return
		}
		c.JSON(http.StatusOK, response)
```

Reply:
Just added returns for failure scenarios.